astropy/stats/lombscargle/_lombscargle.py

```diff
@@ -0,0 +1,255 @@
+"""Main Lomb-Scargle Implementation"""
+import numpy as np
+
+from .implementations import lombscargle, available_methods
+from .implementations.mle import periodic_fit
+
+
+class LombScargle(object):
+    """Compute the Lomb-Scargle Periodogram
+
+    Parameters
+    ----------
+    t : array_like or Quantity
+        sequence of observation times
+    y : array_like or Quantity
+        sequence of observations associated with times t
+    dy : float, array_like or Quantity (optional)
+        error or sequence of observational errors associated with times t
+    fit_bias : bool (optional, default=True)
+        if True, include a constant offet as part of the model at each
+        frequency. This can lead to more accurate results, especially in the
+        case of incomplete phase coverage.
+    center_data : bool (optional, default=True)
+        if True, pre-center the data by subtracting the weighted mean
+        of the input data. This is especially important if fit_bias = False
+
+    Examples
+    --------
+    Generate noisy periodic data:
+
+    >>> rand = np.random.RandomState(42)
+    >>> t = 100 * rand.rand(100)
+    >>> y = np.sin(2 * np.pi * t) + rand.randn(100)
+
+    Compute the Lomb-Scargle periodogram on an automatically-determined
+    frequency grid & find the frequency of max power:
+
+    >>> frequency, power = LombScargle(t, y).autopower()
+    >>> frequency[np.argmax(power)]
+    1.0016662310392956
+
+    Compute the Lomb-Scargle periodogram at a user-specified frequency grid:
+
+    >>> freq = np.arange(0.8, 1.3, 0.1)
+    >>> LombScargle(t, y).power(freq)
+    array([ 0.0204304 ,  0.01393845,  0.35552682,  0.01358029,  0.03083737])
+
+    If the inputs are astropy Quantities with units, the units will be
```

Comment:
My personal preference would be to just give an example that uses `Quantity`, since this is the astropy standard, but note that plain arrays are also OK. Ideally, use an example where unit conversion is done, say, using times in seconds, but checking at frequencies in `mHz`.